Mathlib/Algebra/Homology/ExactSequence.lean

```diff
@@ -0,0 +1,160 @@
+    ShortComplex C :=
+  ShortComplex.mk (S.map' i j) (S.map' j k) (hS.zero' i j k)
+
+/-- The short complex consisting of maps `S.map' i (i + 1)` and `S.map' (i + 1) (i + 2)`
+when we know that `S : ComposableArrows C n` satisfies `S.IsComplex`. -/
+abbrev sc (hS : S.IsComplex) (i : ℕ) (hi : i + 2 ≤ n := by linarith) :
+    ShortComplex C :=
+    S.sc' hS i (i + 1) (i + 2)
+
+/-- `F : ComposableArrows C n` is exact if it is a complex and that all short
+complexes consisting of two consecutive arrows are exact. -/
+structure Exact extends S.IsComplex : Prop where
+  exact (i : ℕ) (hi : i + 2 ≤ n := by linarith) : (S.sc toIsComplex i).Exact
+
+variable {S}
+
+lemma IsExact.exact' (hS : S.Exact) (i j k : ℕ) (hij : i + 1 = j := by linarith)
+    (hjk : j + 1 = k := by linarith) (hk : k ≤ n := by linarith) :
+    (S.sc' hS.toIsComplex i j k).Exact := by
+  subst hij hjk
+  exact hS.exact i hk
+
+/-- Functoriality maps for `ComposableArrows.sc'`. -/
+@[simps]
+def sc'Map {S₁ S₂ : ComposableArrows C n} (φ : S₁ ⟶ S₂) (h₁ : S₁.IsComplex) (h₂ : S₂.IsComplex)
+    (i j k : ℕ) (hij : i + 1 = j := by linarith)
+    (hjk : j + 1 = k := by linarith) (hk : k ≤ n := by linarith) :
+    S₁.sc' h₁ i j k ⟶ S₂.sc' h₂ i j k where
+  τ₁ := φ.app _
+  τ₂ := φ.app _
+  τ₃ := φ.app _
+
+/-- Functoriality maps for `ComposableArrows.sc`. -/
+@[simps!]
+def scMap {S₁ S₂ : ComposableArrows C n} (φ : S₁ ⟶ S₂) (h₁ : S₁.IsComplex) (h₂ : S₂.IsComplex)
+    (i : ℕ) (hi : i + 2 ≤ n := by linarith) :
+    S₁.sc h₁ i ⟶ S₂.sc h₂ i :=
+  sc'Map φ h₁ h₂ i (i + 1) (i + 2)
+
+/-- The isomorphism `S₁.sc' _ i j k ≅ S₂.sc' _ i j k` induced by an isomorphism `S₁ ≅ S₂`
+in `ComposableArrows C n`. -/
+@[simps]
+def sc'MapIso {S₁ S₂ : ComposableArrows C n} (e : S₁ ≅ S₂)
+    (h₁ : S₁.IsComplex) (h₂ : S₂.IsComplex) (i j k : ℕ) (hij : i + 1 = j := by linarith)
+    (hjk : j + 1 = k := by linarith) (hk : k ≤ n := by linarith) :
+    S₁.sc' h₁ i j k ≅ S₂.sc' h₂ i j k where
+  hom := sc'Map e.hom h₁ h₂ i j k
+  inv := sc'Map e.inv h₂ h₁ i j k
+  hom_inv_id := by ext <;> dsimp <;> simp
+  inv_hom_id := by ext <;> dsimp <;> simp
+
+/-- The isomorphism `S₁.sc _ i ≅ S₂.sc _ i` induced by an isomorphism `S₁ ≅ S₂`
+in `ComposableArrows C n`. -/
+@[simps]
+def scMapIso {S₁ S₂ : ComposableArrows C n} (e : S₁ ≅ S₂)
+    (h₁ : S₁.IsComplex) (h₂ : S₂.IsComplex)
+    (i : ℕ) (hi : i + 2 ≤ n := by linarith) :
+    S₁.sc h₁ i ≅ S₂.sc h₂ i where
+  hom := scMap e.hom h₁ h₂ i
+  inv := scMap e.inv h₂ h₁ i
+  hom_inv_id := by ext <;> dsimp <;> simp
+  inv_hom_id := by ext <;> dsimp <;> simp
+
+lemma exact_of_iso {S₁ S₂ : ComposableArrows C n} (e : S₁ ≅ S₂) (h₁ : S₁.Exact) :
+    S₂.Exact where
+  toIsComplex := isComplex_of_iso e h₁.toIsComplex
+  exact i hi := ShortComplex.exact_of_iso (scMapIso e h₁.toIsComplex
+    (isComplex_of_iso e h₁.toIsComplex) i) (h₁.exact i hi)
+
+lemma exact_iff_of_iso {S₁ S₂ : ComposableArrows C n} (e : S₁ ≅ S₂) :
+    S₁.Exact ↔ S₂.Exact :=
+  ⟨exact_of_iso e, exact_of_iso e.symm⟩
+
```

Comment:
Could you please add a statement that if you start with a `ShortComplex` and turn it into a `ComposableArrows _ 2`, then one is exact iff the other is exact?

Reply:
This is planned for the next PR #8159 (draft)!